### ISSUE TITLE
Adapt to removal of --slave in R 4.0

### ISSUE THREAD

**Issue by jmcphers** [rstudio/rstudio#5923]
R 4.0 removes the `--slave` command line option, replacing it with the apparently synonymous `--no-echo`. 

https://github.com/wch/r-source/commit/f1ff49e74593341c74c20de9517f31a22c8bcb04

This is unfortunate because we use `--slave` in dozens of places. We will need to detect whether we are using R 4.0, and, if we are, pass `--no-echo` instead of `--slave`. 

We will need to do this just to compile successfully under R 4.0, as we use R internals like `R_Slave` that have been renamed in R 4.0.

https://github.com/rstudio/rstudio/blob/7626365859391743fd5f947f45ae654360eff31a/src/cpp/r/session/REmbeddedPosix.cpp#L106-L108

**Comment by rhurlin** [comment]
There are two places, were we would have to replace 'R_Slave'
```
#ag R_Slave
src/cpp/r/session/REmbeddedPosix.cpp
107:   Rp->R_Slave = FALSE ;

src/cpp/r/session/REmbeddedWin32.cpp
139:   pRP->R_Slave = FALSE;
```

and 17 files with 29 places were to replace '--slave'
```
#ag '\-\-slave'
cmake/modules/FindLibR.cmake
35:         COMMAND ${LIBR_EXECUTABLE} "--slave" "--vanilla" "-e"
"cat(R.home())"
59:            COMMAND ${LIBR_EXECUTABLE} "--slave" "--vanilla" "-e"
"cat(R.home())"
70:            COMMAND ${LIBR_EXECUTABLE} "--slave" "--no-save" "-e"
"cat(R.home('include'))"
81:            COMMAND ${LIBR_EXECUTABLE} "--slave" "--no-save" "-e"
"cat(R.home('doc'))"
92:            COMMAND ${LIBR_EXECUTABLE} "--slave" "--no-save" "-e"
"cat(R.home('lib'))"

dependencies/windows/install-boost.cmd
7:R --vanilla --slave -f install-boost.R --args debug static
8:R --vanilla --slave -f install-boost.R --args release static

dependencies/windows/install-openssl.cmd
4:R --vanilla --slave -f install-openssl.R

src/cpp/CMakeLists.txt
384:             COMMAND "${LIBR_EXECUTABLE}" "--vanilla" "--slave" "-e"
"cat(as.character(getRversion()))"

src/cpp/core/r_util/REnvironmentPosix.cpp
470:     " --slave --vanilla -e \"cat(paste("
780:      " --slave --vanilla -e
'cat(R.Version()$major,R.Version()$minor, sep=\".\")'",

src/cpp/rstudio-tests.in
76:   export R_HOME=$(R --slave --vanilla -e
"cat(paste(R.home('home'),sep=':'))")
77:   export R_DOC_DIR=$(R --slave --vanilla -e
"cat(paste(R.home('doc'),sep=':'))")

src/cpp/session/SessionAsyncRProcess.cpp
74:   args.push_back("--slave");

src/cpp/session/SessionModuleContext.cpp
1732:   args.push_back("--slave");

src/cpp/session/modules/SessionDependencies.cpp
372:   args.push_back("--slave");

src/cpp/session/modules/build/SessionBuild.cpp
571:      cmd << "--slave";
985:      args.push_back("--slave");
1114:      cmd << "--slave";
1123:         "    system(paste(shQuote('%2%'), '--vanilla --slave -f',
shQuote(x)))"
1154:      cmd << "--slave";
1240:      cmd << "--slave";

src/cpp/session/modules/tex/SessionRnwWeave.cpp
105:      args.push_back("--slave");

src/cpp/session/modules/SessionRPubs.cpp
149:      args.push_back("--slave");

src/cpp/session/modules/SessionTests.cpp
130:   args.push_back("--slave");

src/cpp/session/modules/SessionUpdates.cpp
80:   args.push_back("--slave");

src/cpp/session/modules/clang/RCompilationDatabase.cpp
564:   args.push_back("--slave");

src/cpp/session/modules/connections/SessionConnections.cpp
656:   args.push_back("--slave");

src/cpp/session/modules/presentation/SlideRequestHandler.cpp
315:   args.push_back("--slave");
```

Perhaps something for a string constant or a function?

**Comment by jmcphers** [comment]
I haven't been able to find an official release date for R 4.0 yet, but one source puts it at Feb 29, which is before the release of 1.3. We will therefore need to backport this to the 1.2 branch.

**Comment by jmcphers** [comment]
It looks like `--slave` is deprecated in R 4.0 but might still be accepted? Empirically we can use it without issue since it hasn't been removed yet:

https://github.com/wch/r-source/blob/06185bd431beb9b6e0883193273d1779ef2c6a2c/src/main/CommandLineArgs.c#L151-L153

**Comment by kevinushey** [comment]
In addition, while `R_Slave` has been renamed to `R_NoEcho` it's fortunately still just the same struct layout so our existing code should still work.

We may need to copy or tweak some struct definitions if we want to compile against R 4.0, though. Or do something hacky like reference the struct by offset rather than name.

**Comment by jmcphers** [comment]
Since `--slave` is just deprecated (not removed) in R 4.0, we probably don't have to fix this for the first release of 1.3, and I've taken it out of the milestone.

We DO need to be able to compile against 4.0 since that's what dev machines will probably start having by default in a few months. I have opened a PR here for that change:

https://github.com/rstudio/rstudio/pull/6017

**Comment by rhurlin** [comment]
Thanks for the information. With the next release of RStudio, I will try against R-devel (aka forthcoming R-4.0) on FreeBSD :)
